lib/memo_wise.rb

```diff
@@ -3,6 +3,46 @@
 require "memo_wise/version"
 
 module MemoWise
-  class Error < StandardError; end
-  # Your code goes here...
+  def initialize(*values)
+    @_memo_wise = {}
+    super
+  end
+
+  def self.prepended(target)
+    class << target
+      # Implements memoization for the given method name.
+      #
+      # @param method_name [Symbol]
+      #   Name of method for which to implement memoization.
+      def memo_wise(method_name)
+        not_memoized_name = "#{method_name}_not_memoized".to_sym
+        alias_method not_memoized_name, method_name
+        private not_memoized_name
+
+        module_eval <<-END_OF_METHOD, __FILE__, __LINE__ + 1
+          def #{method_name}(*args)
+            key = ["#{method_name}", args].freeze
+            @_memo_wise.fetch(key) do
+              @_memo_wise[key] = send(:"#{not_memoized_name}", *args)
+            end
+          end
+        END_OF_METHOD
+      end
+
+      # Undoes memoization for the given method name. If the method was
+      # previously memoized, it will no longer be memoized.
+      #
+      # @param method_name [Symbol]
```

Comment:
How would you feel about also documenting the expected `NoMethodError` if the given method was not previously memoized?  This would be the `@raise` tag in YARD doc ([see here](https://www.rubydoc.info/gems/yard/file/docs/Tags.md#raise)).

I see documenting expected errors as part of a well-maintained API documentation, so I look for parity between all three of:
1. What the code can/will raise in the implementation
2. What the YARD docs say can be raised
3. What the RSpec coverage tests will be raised